src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -55,16 +74,45 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   // Send a user to datastore. Update the current information about the user if one exists.
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String userGitHubID = request.getParameter("githubID");
-    if (userGitHubID.trim().isEmpty()) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid ID");
-      return;
+    String userGitHubAuthToken = request.getParameter("gitHubAuthToken");
+    ArrayList<String> userTags =
+        new ArrayList<>(
+            Arrays.asList(
+                request.getParameterValues("interestTags") == null
+                    ? new String[] {}
+                    : request.getParameterValues("interestTags")));
+
+    String userID;
+    try {
+      userID = AuthServlet.getAuthorizedUser().getUserId();
+    } catch (NullPointerException e) {
+      ErrorResponse.sendJsonError(
+          response,
+          "User not logged in",
+          HttpServletResponse.SC_FORBIDDEN,
+          "You are not logged in");
+      return; // TODO: Establish Redirect page path
     }
-    ArrayList<String> userTags = new ArrayList<>(Arrays.asList(request.getParameterValues("tags")));
 
-    Entity personEntity =
-        new PersonBuilder().gitHubID(userGitHubID).interestTags(userTags).buildPersonEntity();
-    datastore.put(personEntity);
+    // Get User information from GitHub using the Oath token.
+    GHMyself userGHMyself;
+    try {
+      userGHMyself = GitHub.connectUsingOAuth(userGitHubAuthToken).getMyself();
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage()
+              + e.getStackTrace()
+              + "Could not get authenticated User with the given token",
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "User could not be authenticated by GitHub, please try again");
+      return;
+    }
+    // Build and send the User's datastore entity
```

Comment:
Does this mean to say "Build and save," or am I missing something here? This is a bit confusing.

Reply:
Yeah it does. 
```suggestion
    // Build and save the User's datastore entity
```